src/modules/cron/cron.controller.ts

```diff
@@ -53,4 +62,75 @@ export class CronController {
+    @ApiInternalServerErrorResponse({
+        description: 'Internal server error while trying to remove personenkontexte from users.',
+    })
+    public async removePersonenKontexteWithExceedingLimitFromUsers(
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<boolean> {
+        try {
+            //Get PersonenKontexte for a person at least one kontext exceeds the befristung value
+            const personenKontexteGroupedByPersonId: Record<string, Personenkontext<true>[]> =
+                await this.personenKonextRepository.getPersonenKontexteWithExceedingBefristung();
+            if (Object.keys(personenKontexteGroupedByPersonId).length === 0) {
+                return true;
+            }
+
+            //filter each Personenkontext and validate Personenkontexte to keep for each person
+            const promises: Promise<Personenkontext<true>[] | PersonenkontexteUpdateError>[] = [];
+            for (const personId in personenKontexteGroupedByPersonId) {
+                const personenKontexte: Personenkontext<true>[] | undefined =
+                    personenKontexteGroupedByPersonId[personId];
+
+                if (Array.isArray(personenKontexte)) {
+                    const count: number = personenKontexte.length;
+                    //filter PersonenKontexte
+                    const personenKontexteToKeep: DbiamPersonenkontextBodyParams[] =
+                        this.filterPersonenKontexte(personenKontexte);
+                    //validate PersonenKontexte to keep
+                    promises.push(
+                        this.personenkontextWorkflowFactory
+                            .createNew()
+                            .commit(personId, new Date(), count, personenKontexteToKeep, permissions),
+                    );
+                }
+            }
+
+            //validate results
+            const results: PromiseSettledResult<Personenkontext<true>[] | PersonenkontexteUpdateError>[] =
+                await Promise.allSettled(promises);
+            const allSuccessful: boolean = results.every(
+                (result: PromiseSettledResult<Personenkontext<true>[] | PersonenkontexteUpdateError>) =>
+                    result.status === 'fulfilled' && !(result.value instanceof PersonenkontexteUpdateError),
+            );
+
+            return allSuccessful;
+        } catch (error) {
+            throw new Error('Failed to remove kontexte due to an internal server error.');
+        }
+    }
+
+    private filterPersonenKontexte(personenKontexte: Personenkontext<true>[]): DbiamPersonenkontextBodyParams[] {
+        const today: Date = new Date();
+        today.setHours(0, 0, 0, 0);
+        const personenKontexteToKeep: DbiamPersonenkontextBodyParams[] = [];
+        personenKontexte.forEach((personenKontext: Personenkontext<true>) => {
+            if (personenKontext.befristung == undefined || personenKontext.befristung >= today) {
```

Comment:
Don't compare to `undefined`, might cause problems will `null`-values
```suggestion
            if (!personenKontext.befristung || personenKontext.befristung >= today) {
```

Reply:
Obviously, thanks for noticing :D